torch/distributed/_composable/replicate.py

```diff
@@ -16,6 +39,9 @@ def __init__(self) -> None:
 
     def mark_modules(self, *modules: nn.Module, **kwargs) -> None:
         for module in modules:
+            assert _can_compose(
```

Comment:
nit: I am wondering if we should make this an explicit `raise AssertionError` since `assert` may be optimized out.